scripts/register_sam.py

```diff
@@ -15,6 +22,8 @@ def main():
     tdr = TDRClient()
     tdr.register_with_sam()
     tdr.verify_authorization()
+    plugin = Plugin(TDRSource.parse(config.tdr_source(config.default_catalog)))
```

Comment:
Looks brittle. Seems to assume that the default catalog uses the TDR plug-in.

Reply:
Due to the fallback in `tdr_source` It won't break *today* if the default catalog isn't using TDR but its is still odd and may break if the `tdr_source` implementation changes AND the default catalog isn't using TDR. Instead of using the first configured catalog (the default catalog) this should probably test every TDR catalog.

Reply:
My fixup for this will skip the BigQuery access test if no catalogs are configured to use the TDR plugin. This seems sensible to me but it seemed important to point out.